std/container/slist.d

```diff
@@ -135,6 +135,48 @@ struct SList(T)
         return n;
     }
 
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isImplicitlyConvertible!(Stuff, T))
+    {
+        import std.range : only;
+        return createNodeChain(only(stuff));
+    }
+
+    private static auto createNodeChain(Stuff)(Stuff stuff)
+    if (isInputRange!Stuff && isImplicitlyConvertible!(ElementType!Stuff, T))
+    {
+        static struct Chain
+        {
+            Node* first;
+            Node* last;
+            size_t length;
+        }
+
+        Chain ch;
+
+        foreach (item; stuff)
+        {
+            auto newNode = new Node(null, item);
+            (ch.first ? ch.last._next : ch.first) = newNode;
+            ch.last = newNode;
+            ++ch.length;
+        }
```

Comment:
> Also optimize

Technically this is not an optimization because now a bit more work needs to be done in the special case of just inserting one node. I doubt that it matters though (and yes saving the allocation on the root node is a win).